docs/metatype.dev/docs/reference/graphql/index.mdx

```diff
@@ -0,0 +1,183 @@
+---
+sidebar_position: 50
+---
+
+import TGExample from "@site/src/components/TGExample";
+
+# GraphQL
+
+[GraphQL](https://graphql.org/) is the primary means of querying your typegraph. 
+This page documents all the semantics of how your typegraph translates into a GraphQL schema.
+
+## `Query` and `Mutation`
+
+The root functions passed to the `expose` function will be added as fields to the [special `query` and `mutation`](https://graphql.org/learn/schema/#the-query-and-mutation-types) GraphQL objects.
+Under which object a function is assigned depends on it's [effect](/docs/reference/types/functions).
+Note that this assignment still holds for deeply nested functions so be sure to avoid including functions that have mutating effects under query responses or vice versa.
+
+## Variables
+
+While [GraphQL Variables](https://graphql.org/learn/queries/#variables) work as expected for the most part, the typegate currently does not do type validation of variables for the provided query documents.
```

Comment:
Do we already have a ticket for this?

Reply:
This is an intentional property, remember? `metagen` currently relies on this and the check would duplicate the validator check the typegate already does.